File: others/min_val.py
def min_val(l):
    min_idx = 0
    for i in range(len(l) - 1):
        if l[min_idx] > l[i]:
            min_idx = i
    return l[min_idx]


l = [-1, 2, 3, 5, 4, 1, 0, 8]
print(min_val(l))

File: others/min_val.go
package main

import (
	"fmt"
)

func minVal(l []int) int {
	minIdx := 0
	for i := 0; i < len(l)-1; i++ {
		if l[i] < l[minIdx] {
			minIdx = i
		}
	}
	return l[minIdx]
}

func main() {
	l := []int{-1, 4, 3, 1, 2, 0, 5}
	fmt.Println(minVal(l))
}
